splunktalib/rest.py

```
@@ -49,104 +51,45 @@ def splunkd_request(
+                data=data,
+                headers=headers,
+                timeout=timeout,
+                verify=verify,
             )
         except Exception:
             log.logger.error(msg_temp, splunkd_uri, "unknown", format_exc())
         else:
-            if resp.status not in (200, 201):
-                if not (method == "GET" and resp.status == 404):
+            if resp.status_code not in (200, 201):
+                if not (method == "GET" and resp.status_code == 404):
                     log.logger.debug(
-                        msg_temp, splunkd_uri, resp.status, code_to_msg(resp, content)
+                        msg_temp, splunkd_uri, resp.status_code, code_to_msg(resp)
                     )
             else:
-                return resp, content
+                return resp
     else:
-        return resp, content
+        return resp
 
 
-def code_to_msg(resp, content):
+def code_to_msg(response: requests.Response):
     code_msg_tbl = {
-        400: "Request error. reason={}".format(content),
+        400: "Request error. reason={}".format(response.text),
         401: "Authentication failure, invalid access credentials.",
         402: "In-use license disables this feature.",
         403: "Insufficient permission.",
         404: "Requested endpoint does not exist.",
-        409: "Invalid operation for this endpoint. reason={}".format(content),
-        500: "Unspecified internal server error. reason={}".format(content),
+        409: "Invalid operation for this endpoint. reason={}".format(response.text),
+        500: "Unspecified internal server error. reason={}".format(response.text),
         503: (
             "Feature is disabled in the configuration file. "
-            "reason={}".format(content)
+            "reason={}".format(response.text)
         ),
     }
 
-    return code_msg_tbl.get(resp.status, content)
-
-
-def build_http_connection(config, timeout=120, disable_ssl_validation=False):
```

For this removal, can we raise an exception or mention it in the README docs somewhere so that other users know about this behavior?

---

We can create a Github issue in this repo, document this removal and say that it is going to be a part of `splunktalib` 3.0.0 release.

---

@mchavda-splunk can you please create something similar to https://github.com/splunk/addonfactory-ta-library-python/issues/38 but describing things that are going to be removed as part of this PR?

---

@arys-splunk  Created #50. Let me know if you have any suggestions.

---

The issue looks good to me.